staging/src/k8s.io/api/flowcontrol/v1alpha1/types.go

```diff
@@ -0,0 +1,414 @@
+}
+
+type SubjectKind string
+
+// Supported subject's kinds.
+const (
+	SubjectKindUser           SubjectKind = "User"
+	SubjectKindGroup          SubjectKind = "Group"
+	SubjectKindServiceAccount SubjectKind = "ServiceAccount"
+)
+
+// UserSubject holds detailed information for user-kind subject.
+type UserSubject struct {
+	// Required
+	Name string `json:"name" protobuf:"bytes,1,opt,name=name"`
+}
+
+// GroupSubject holds detailed information for group-kind subject.
+type GroupSubject struct {
+	// Required
+	Name string `json:"name" protobuf:"bytes,1,opt,name=name"`
+}
+
+// ServiceAccountSubject holds detailed information for service-account-kind subject.
+type ServiceAccountSubject struct {
+	// Required
+	Namespace string `json:"namespace" protobuf:"bytes,1,opt,name=namespace"`
+	// Required
+	Name string `json:"name" protobuf:"bytes,2,opt,name=name"`
+}
+
+// ResourcePolicyRule is a predicate that matches some resource requests, testing the request's verb and the resource
+// that the request seeks to act upon. A ResourcePolicyRule matches a request if and only if: (a) at least one member
+// of verbs matches the request, (b) at least one member of apiGroups matches the request, and (c) at least one member
+// of resources matches the request.
+type ResourcePolicyRule struct {
+	// `verbs` is a list of matching verbs and may not be empty.
+	// "*" represents all verbs.
+	// Required.
+	Verbs []string `json:"verbs" protobuf:"bytes,1,rep,name=verbs"`
+	// `apiGroups` is a list of matching API groups and may not be empty.
+	// "*" represents all api-groups.
+	// Required.
+	APIGroups []string `json:"apiGroups" protobuf:"bytes,2,rep,name=apiGroups"`
+	// `resources` is a list of matching resource (i.e., lowercase and plural) with, if desired, subresource. This list
+	// may not be empty.
+	// "*" represents all resources.
+	// Required.
+	Resources []string `json:"resources" protobuf:"bytes,3,rep,name=resources"`
+}
+
+// NonResourcePolicyRule is a predicate that matches non-resource requests according to their verb and the non-resource
+// URL they request to act upon. A NonResourcePolicyRule matches a request if and only if both (a) at least one member
+// of verbs matches the request and (b) at least one member of nonResourceURLs matches the request.
+type NonResourcePolicyRule struct {
+	// `verbs` is a list of matching verbs and may not be empty.
+	// "*" represents all verbs.
+	// Required.
+	Verbs []string `json:"verbs" protobuf:"bytes,1,rep,name=verbs"`
+	// `nonResourceURLs` is a set of partial urls that a user should have access to.
+	// "*" represents all non-resource urls.
+	// Required.
+	NonResourceURLs []string `json:"nonResourceURLs" protobuf:"bytes,6,rep,name=nonResourceURLs"`
+}
+
+// FlowSchemaStatus represents the current state of a flow-schema.
+type FlowSchemaStatus struct {
+	// `conditions` is a list of current states of flow-schema.
```

Comment:
s/of/of the/

s/flow-schema/FlowSchema/

Reply:
updated. and replaced all `flow-schema` w/ camel-case `FlowSchema`